tests/e2e/kcs/test_selinux_relabel_solution.py

```diff
@@ -330,3 +330,114 @@ def test_selinux_relabel_for_existing_pvc(
+    def test_selinux_relabel_for_new_pvc(
+        self,
+        pvc_factory,
+        service_account_factory,
+        storageclass_factory,
+        copies,
+    ):
+        """
+        Steps:
+            1. Create storageclass with security context set.
+            2. Create cephfs pvcs and attach pod with more than 100K files across multiple nested directories
+            3. Take md5sum for them some random files and get pod restart time
+            4. Restart the pod which is hosting cephfs files in large numbers.
+            5. Check data integrity.
+            6. Check for relabeling - this should not be happening.
+
+        Args:
+            pvc_factory (function): A call to pvc_factory function
+            service_account_factory (function): A call to service_account_factory function
+            storageclass_factory ((function): A call to storageclass_factory function
+            copies (int): number of copies to write kernel files in pod
+
+        """
+        # Create storageclass with security context
+        self.ocp_project = ocp.OCP(
+            kind=constants.NAMESPACE, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        sc_name = "ocs-storagecluster-cephfs-selinux-relabel"
+        self.storage_class = storageclass_factory(
+            sc_name=sc_name,
+            interface=constants.CEPHFILESYSTEM,
+            kernelMountOptions='context="system_u:object_r:container_file_t:s0"',
+        )
+
+        # Create PVC using new storageclass
+        self.pvc_obj = pvc_factory(
+            project=self.ocp_project,
+            storageclass=self.storage_class,
+            size="20",
+        )
+        log.info(f"PVC {self.pvc_obj.name} created")
+
+        # Create service_account to get privilege for deployment pods
+        self.service_account_obj = service_account_factory(
+            project=self.ocp_project,
+        )
+
+        # Create deployment pod
+        self.pod_obj = self.create_deploymentconfig_pod(
+            command=["/opt/multiple_files.sh"],
+            command_args=[f"{copies}", "/mnt"],
+        )
+
+        log.info(f"pod {self.pod_obj.name} created")
+        self.pod_selector = self.pod_obj.labels.get(constants.DEPLOYMENTCONFIG)
+
+        # Get the md5sum of some random files
+        data_path = f"{constants.FLEXY_MNT_CONTAINER_DIR}"
+        num_of_files = random.randint(3, 9)
+        pod = ocp.OCP(kind="pod", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+        random_files = pod.exec_oc_cmd(
+            f"exec -it {self.pod_obj.name} -- /bin/bash"
+            f' -c "find {data_path} -type f | "shuf" -n {num_of_files}"',
+            timeout=300,
+        )
+        random_files = random_files.split()
+        log.info(f"files are {random_files}")
+        initial_md5sum = []
+        for file_path in random_files:
+            md5sum = res_pod.cal_md5sum(
+                pod_obj=self.pod_obj,
+                file_name=file_path,
```

Comment:
the same code is present even in the first test, maybe you can create a common function and call it in both the tests which reduces the code redundancy